### PR TITLE
Add failing test for demo prep

### PR DESCRIPTION
**Pull request by jbolinger**
I would like to be able to have folks do something along the lines #44 for the demo next week. They should be able to build an API (showcase in particular) using just a simple Gemfile/Rakefile pair that will take care of installing everything beyond Ruby and bundler. We'll also show how to write a simple script to use the generated library (and we'll likely have rake run it after generation so it's a one liner demo) i.e.:

```ruby
require "google/showcase/v1alpha3/echo"

client = Google::Showcase::V1alpha3::EchoClient.new(
  credentials: GRPC::Core::Channel.new(
    "localhost:7469", nil, :this_channel_is_insecure))

response = client.echo 'hi there!', nil

puts response.content
```

However, we need to get this test that I've added to pass in order to do that. I was able to edit the generated files by hand to get this test to pass without too much trouble, so I think we can make this happen in time.

To be clear, I don't want to use this integration test in this PR for the demo directly. We'll create a very simple standalone Ruby project that will be very similar. But, we have to get this test to pass for the demo to actually work.

So, let's try to get this test in quickly and then jump on to the harder part - getting it to pass 😄 .

### CODE REVIEW

**Comment by blowmage** [comment]
> I was able to edit the generated files by hand to get this test to pass

What edits to the generated files are you making?

File: shared/Rakefile [hunk 1]
```diff
@@ -65,17 +65,31 @@ end
...
-  t.test_files = FileList["test/**/*_test.rb"]
-  t.warning = false
+task :test do
+  Rake::Task["test:showcase"].invoke
 end
 namespace :test do
   desc "Run functional tests for showcase protos"
   Rake::TestTask.new :showcase do |t|
-    t.libs << "test"
+    require "tmpdir"
```
Comment (blowmage) [inline shared/Rakefile]:
```suggestion
    require "minitest"
    require "fileutils"
    require "tmpdir"
```

File: shared/Rakefile [hunk 2]
```diff
@@ -65,17 +65,31 @@ end
...
+      "--ruby_gapic_out=#{client_lib}",
+      "protos/showcase/v1alpha3/echo.proto"
+    ].join " "
+    sh protoc_cmd
+
+    t.libs << ["test",  client_lib]
     t.test_files = FileList["test/showcase/**/*_test.rb"]
     t.warning = false
+
+    # TODO: delete the temporary dir?
```
Comment (blowmage) [inline shared/Rakefile]:
```suggestion
    Minitest.after_run do
      FileUtils.remove_dir client_lib, true
    end
```

File: shared/Rakefile [hunk 3]
```diff
@@ -65,17 +65,31 @@ end
...
+task :test do
+  Rake::Task["test:showcase"].invoke
 end
 namespace :test do
   desc "Run functional tests for showcase protos"
   Rake::TestTask.new :showcase do |t|
-    t.libs << "test"
+    require "tmpdir"
+
+    client_lib = Dir.mktmpdir
```
Comment (blowmage) [inline shared/Rakefile]: I think creating the temp directory, running the `protoc` command, and cleaning up the temp directory could all be done in the test helper. The test helper already starts and stops the echo server. I don't think this _needs_ to be performed in the rake task.

Comment (jbolinger) [reply shared/Rakefile]: Yeah, that's true. I'll move it.

**Comment by jbolinger** [comment]
@blowmage I put the client that I got the test to pass with in #46. Note that it's just an example and there is no reason that we have to do it exactly like that. I just want to get us in a state where the client is runnable.

File: shared/Rakefile (hunk 2 above)
Comment (jbolinger) [reply shared/Rakefile]: I added this when I moved it to the helper. Tx!

File: shared/Rakefile (hunk 1 above)
Comment (jbolinger) [reply shared/Rakefile]: Added in the helper.

**Pull request merged**